src/components/AbilityFilter.tsx

```diff
@@ -1,44 +1,41 @@
 import axios from "axios";
-import { useState, useEffect } from "react";
-import { useQuery } from "@tanstack/react-query";
 import { Spinner } from "@chakra-ui/react";
 import DesktopPokemonList from "./DesktopPokemonList";
+import { useInfiniteQuery } from "@tanstack/react-query";
 
 interface Props {
   choice: string;
-  offset: number;
-
-  setPages: (pages: number) => void;
 }
 
 function AbilityFilter(props: Props) {
-  let pokemonArray: pokemon[] = [];
-
-  const [pokemon, setPokemon] = useState<pokemon[]>(pokemonArray);
-
-  const { isLoading, error, data } = useQuery(
-    ["ability", props.offset, props.choice],
-    async () => {
-      const response = await axios.get(
-        `http://localhost:8081/api/pokemon/ability?ability=${props.choice}&offset=${props.offset}&pageSize=24`
-      );
-      const data = await response.data;
-      return data;
+  const fetch = ({ pageParam: offset = 1 }) =>
+    axios.get(
+      `http://localhost:8081/api/pokemon/ability?ability=${props.choice}&offset=${offset}&pageSize=24`
+    );
+
+  const { isLoading, error, data, fetchNextPage } = useInfiniteQuery(
+    ["content", props.choice],
+    fetch,
+    {
+      getNextPageParam: (lastPage, pages) => {
+        if (!lastPage.data.last) {
```

Comment:
I would invert the logic here. Check for failure, and return the 'happy path' as the bottom-most statement.